packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -0,0 +1,820 @@
+      throw new Error('Invalid response')
+
+    const events = await this.getTransactionEvents({
+      txHash,
+      eventTopics: this.eventTopics.ownershipTransferred,
+    })
+    const event = events.length > 0 ? events[0] : undefined
+    if (event) {
+      return {
+        event,
+      }
+    }
+
+    throw new TransactionFailedError()
+  }
+
+  async setPause(setPausedArgs: SetPausedConfig): Promise<{
+    event: Log
+  }> {
+    const txHash = await this._setPaused(setPausedArgs)
+
+    if (!this._isContractTransaction(txHash))
+      throw new Error('Invalid response')
+
+    const events = await this.getTransactionEvents({
+      txHash,
+      eventTopics: this.eventTopics.setPaused,
+    })
+    const event = events.length > 0 ? events[0] : undefined
+    if (event) {
+      return {
+        event,
+      }
+    }
+
+    throw new TransactionFailedError()
+  }
+
+  async execCalls(execCallsArgs: SplitV2ExecCallsConfig): Promise<{
+    event: Log
+  }> {
+    const txHash = await this._execCalls(execCallsArgs)
+
+    if (!this._isContractTransaction(txHash))
+      throw new Error('Invalid response')
+
+    const events = await this.getTransactionEvents({
+      txHash,
+      eventTopics: this.eventTopics.execCalls,
+    })
+    const event = events.length > 0 ? events[0] : undefined
+    if (event) {
+      return {
+        event,
+      }
+    }
+
+    throw new TransactionFailedError()
+  }
+
+  async distribute(distributeArgs: DistributeSplitConfig): Promise<{
+    event: Log
+  }> {
+    const txHash = await this._distribute(distributeArgs)
+
+    if (!this._isContractTransaction(txHash))
+      throw new Error('Invalid response')
+
+    const events = await this.getTransactionEvents({
+      txHash,
+      eventTopics: this.eventTopics.splitDistributed,
+    })
+    const event = events.length > 0 ? events[0] : undefined
+    if (event) {
+      return {
+        event,
+      }
+    }
+
+    throw new TransactionFailedError()
+  }
+
+  async updateSplit(updateSplitArgs: UpdateSplitV2Config): Promise<{
+    event: Log
+  }> {
+    const txHash = await this._updateSplit(updateSplitArgs)
+
+    if (!this._isContractTransaction(txHash))
+      throw new Error('Invalid response')
+
+    const events = await this.getTransactionEvents({
+      txHash,
+      eventTopics: this.eventTopics.splitUpdated,
+    })
+    const event = events.length > 0 ? events[0] : undefined
+    if (event) {
+      return {
+        event,
+      }
+    }
+
+    throw new TransactionFailedError()
+  }
+
+  async predictDeterministicAddress(
+    createSplitArgs: CreateSplitV2Config,
+  ): Promise<{
+    splitAddress: Address
+  }> {
+    validateAddress(createSplitArgs.owner)
+    createSplitArgs.recipients.map((recipient) => validateAddress(recipient))
+    this._requirePublicClient()
+
+    const factory = this._getSplitV2FactoryContract(createSplitArgs.splitType)
+
+    const totalAllocation = createSplitArgs.allocations.reduce((a, b) => a + b)
```

Comment:
nit: i think this is the same as the create split transaction function? maybe some helper function to get the args? just so we're not repeating stuff twice (like i think this `reduce` also needs the starting point? unless it defaults to 0 or something, wasn't aware of that).